draft-bishop-httpbis-http2-additional-certs.md

```diff
@@ -643,16 +639,17 @@ received on any other stream MUST be rejected with a stream error of type
 ### Exported Authenticator Characteristics {#exp-auth}
 
 The Exported Authenticator API defined in [I-D.ietf-tls-exported-authenticator]
-takes as input a certificate, supporting information about the certificate
-(OCSP, SCT, etc.), and an optional `certificate_request_context`.  When
-generating exported authenticators for use with this extension, the
-`certificate_request_context` MUST be the two-octet Cert-ID.
-
-Upon receipt of a completed authenticator, an endpoint MUST check that:
-
- - the `validate` API confirms the validity of the authenticator itself
- - the `certificate_request_context` matches the Cert-ID of the frame(s) in
-   which it was received
+takes as input a request, a set of certificates, and supporting information
+about the certificate (OCSP, SCT, etc.).
+
+Upon receipt of a completed authenticator, an endpoint MUST perform the
+following steps:
+ - Using the `get context` API, retrieve the `certificate_request_context` used
+   to generate the authenticator, if any.
+ - Verify that the `certificate_request_context` is one previously generated or,
+   when processed by a client, is absent.
+ - Use the `validate` API to confirm the validity of the authenticator with
+   regard to the generated request.
```

Comment:
I think you can simply say that you can validate a spontaneous server-generated CERTIFICATE without using the "get context" api.

Reply:
I'm not certain it's always possible to tell whether the CERTIFICATE frame is spontaneous or not.  It's obvious if you don't have an outstanding request, but it's also possible to get a spontaneous certificate after sending a request, and you won't know that until you check the request context in the authenticator.

Always checking the request context should work in all situations, so it seems like we should just do that rather than having more conditionals.  Welcome to convince me otherwise.  ;-)

Reply:
This works, it's just that the "regard to the generated request" is dependent on the result of whether or not a matching CERTIFICATE_REQUEST was found or not. Maybe updating the text to say "regard to the generated request, if a matching request was found"

Reply:
Already done.  The commit from a few minutes ago adds "(if any)".